operator/pkg/gateway-api/gateway_reconcile.go

```diff
@@ -172,13 +186,16 @@ func (r *gatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	return controllerruntime.Success()
 }
 
-func (r *gatewayReconciler) ensureService(ctx context.Context, desired *corev1.Service) error {
+func (r *gatewayReconciler) ensureService(ctx context.Context, desired *corev1.Service, gw *gatewayv1.Gateway) error {
 	svc := desired.DeepCopy()
 	_, err := controllerutil.CreateOrPatch(ctx, r.Client, svc, func() error {
 		// Save and restore loadBalancerClass
 		// e.g. if a mutating webhook writes this field
 		lbClass := svc.Spec.LoadBalancerClass
-
+		if len(gw.Spec.Addresses) != 0 {
+			// This annotation must be set through gateway.spec.addresses, otherwise it will be deleted.
+			delete(svc.Annotations, annotation.LBIPAMIPKeyAlias)
+		}
```

Comment:
This is actually a breaking change, because if someone was setting this annotation on a created Service resource manually, it will now be removed.

I'd suggest we log a warning about deprecated label use and say that this will change in the future to removing this label if no `spec.Addresses` are set.

Something like:
```
DEPRECATED: The service <servnicenamespace/<servicename> is setting an IP address using the annotation <annotation>. These should be set using the spec.addresses field in Gateway objects instead. At a future date this annotation will be removed from Gateway-created Services if no spec.addresses are set.
```

That's a long message though, suggestions welcomed.

Then we need a followup issue to come back and change this to delete the annotation instead.

Reply:
What if I use the infrastructure stanza for this, will that also not be supported anymore?

Reply:
If we make this remove those annotations, then yes, there would be a problem. Practically, I think that if we do make the breaking change here, we will need to check the labels and annotations and do something (either put a warning in the status, or reject outright) Gateways that use infrastructure to set this label.

Really good point, thanks for that.